src/core/source_buffers/segment_inventory.ts

```diff
@@ -655,3 +619,166 @@ export default class SegmentInventory {
     return this.inventory;
   }
 }
+
+/**
+ * Returns `true` if the buffered start of the given chunk looks coherent enough
+ * relatively to what is anounced in the Manifest.
+ * @param {Object} thisSegment
+ * @returns {Boolean}
+ */
+function bufferedStartLooksCoherent(
+  thisSegment : IBufferedChunk
+) : boolean {
+  if (thisSegment.bufferedStart === undefined ||
+      !thisSegment.isCompleteSegment)
+  {
+    return false;
```

Comment:
If the segment is not complete, it may have a start that will not change when the segment will be complete.
So, maybe we shouldn't return `false`.